### PR TITLE
Update RELEASES.md for 1.41.0

### PR DESCRIPTION
**Pull request by XAMPPRocky**
### [Rendered](https://github.com/XAMPPRocky/rust/blob/relnotes-1.41.0/RELEASES.md)

cc @rust-lang/release

### CODE REVIEW

File: RELEASES.md [hunk 1]
```diff
@@ -1,3 +1,109 @@
+Version 1.41.0 (2020-01-30)
+===========================
+
+Language
+--------
+
+- [You can now pass type parameters to foreign items when implementing
+  traits.][65879] E.g. You can now write `impl<T> From<Foo> for Vec<T> {}`.
+- [You can now use any receiver in the `self` position.][64325] E.g. you can
```
Comment (Centril) [inline RELEASES.md]:
```suggestion
- [You can now arbitrarily nest receiver types in the `self` position.][64325] E.g. you can
```

File: RELEASES.md [hunk 2]
```diff
@@ -1,3 +1,109 @@
...
+  now write `fn foo(self: Box<Box<Self>>) {}`. Previously only `Self`, `&Self`,
+  `&mut Self`, `Arc<Self>`, `Rc<Self>`, and `Box<Self>` were allowed.
+- [You can now use any valid identifier in a `format_args` macro.][66847]
+  Previously this was a strict subset.
+- [You can now use `!`/"Never type" in any type position.][65355]
```
Comment (Centril) [inline RELEASES.md]:
```suggestion
```
twas reverted... again :(

File: RELEASES.md [hunk 3]
```diff
@@ -1,3 +1,109 @@
...
+
+Compiler
+--------
+
+- [Rustc's parser now allows visibility in trait items and enum variants.][66183]
+  These are still rejected when compiled, however they can be seen and parsed
+  by procedural macros.
```
Comment (Centril) [inline RELEASES.md]: This is a language change, move it up.

File: RELEASES.md [hunk 4]
```diff
@@ -1,3 +1,109 @@
...
+- [You can now provide `--extern` flag without a path, indicating that it is
+  available from the search path or specified with an `-L` flag.][64882]
+
+\* Refer to Rust's [platform support page][forge-platform-support] for more
+information on Rust's tiered platform support.
+
+Libraries
+---------
+
+- [The `core::panic` module is now stable.][66771] This was already stable
```
Comment (Centril) [inline RELEASES.md]:
```suggestion
- [The `core::panic` module is now stable.][66771] It was already stable
```

File: RELEASES.md [hunk 5]
```diff
@@ -1,3 +1,109 @@
...
+  Previously this was a strict subset.
+- [You can now use `!`/"Never type" in any type position.][65355]
+
+Compiler
+--------
+
+- [Rustc's parser now allows visibility in trait items and enum variants.][66183]
+  These are still rejected when compiled, however they can be seen and parsed
+  by procedural macros.
+- [Rustc will now warn if you have unused `'loop` labels.][66325]
```
Comment (Mark-Simulacrum) [inline RELEASES.md]:
```suggestion
- [Rustc will now warn if you have unused loop `'label`s.][66325]
```

File: RELEASES.md [hunk 6]
```diff
@@ -1,3 +1,109 @@
...
+- [Removed support for the `i686-unknown-dragonfly` target.][67255]
+- [Added tier 3 support\* for the `riscv64gc-unknown-linux-gnu` target.][66661]
+- [You can now pass an arguments file passing the `@path` syntax
+  to rustc.][66172]
```
Comment (Mark-Simulacrum) [inline RELEASES.md]:
```suggestion
  to rustc.][66172]
    Note that the format differs somewhat from what is found in other tooling; see [docs](https://doc.rust-lang.org/nightly/rustc/command-line-arguments.html#path-load-command-line-flags-from-a-path) for more information.
```

(Basically I would like there to be a link to the docs here, but I don't care too much how we accomplish that).

File: RELEASES.md [hunk 7]
```diff
@@ -1,3 +1,109 @@
...
+  optimisation level to `2` for debug builds. You can also use
+  `[profile.<profile>.build_overrides]` to override build scripts and
+  their dependencies.
+
+Misc
+----
+
+- [You can now specify `edition` in documentation code blocks to only test code
+  for that edition.][66238] E.g. `edition2018` will only test code when compiled
+  with the 2018 edition.
```
Comment (Mark-Simulacrum) [inline RELEASES.md]: Hm, did we really stabilize that? I would expect `edition2018` to *compile* with the 2018 edition, not arbitrarily ignore tests depending on the "global" edition rustdoc is using?

File: RELEASES.md [hunk 8]
```diff
@@ -1,3 +1,111 @@
...
+--------
+
+- [You can now pass type parameters to foreign items when implementing
+  traits.][65879] E.g. You can now write `impl<T> From<Foo> for Vec<T> {}`.
+- [You can now arbitrarily nest receiver types in the `self` position.][64325] E.g. you can
+  now write `fn foo(self: Box<Box<Self>>) {}`. Previously only `Self`, `&Self`,
+  `&mut Self`, `Arc<Self>`, `Rc<Self>`, and `Box<Self>` were allowed.
+- [You can now use any valid identifier in a `format_args` macro.][66847]
+  Previously this was a strict subset.
+- [Rustc's parser now allows visibility items (e.g. `pub`) in trait items and
```
Comment (Centril) [inline RELEASES.md]:
```suggestion
- [Visibility modifiers (e.g. `pub`) are now syntactically allowed on trait items and
```

File: RELEASES.md [hunk 9]
```diff
@@ -1,3 +1,111 @@
...
+Language
+--------
+
+- [You can now pass type parameters to foreign items when implementing
+  traits.][65879] E.g. You can now write `impl<T> From<Foo> for Vec<T> {}`.
+- [You can now arbitrarily nest receiver types in the `self` position.][64325] E.g. you can
+  now write `fn foo(self: Box<Box<Self>>) {}`. Previously only `Self`, `&Self`,
+  `&mut Self`, `Arc<Self>`, `Rc<Self>`, and `Box<Self>` were allowed.
+- [You can now use any valid identifier in a `format_args` macro.][66847]
+  Previously this was a strict subset.
```
Comment (jonas-schievink) [inline RELEASES.md]:
```suggestion
  Previously identifiers starting with an underscore were not allowed.
```

File: RELEASES.md [hunk 10]
```diff
@@ -1,3 +1,111 @@
...
+- [Rustc's parser now allows visibility items (e.g. `pub`) in trait items and
+  enum variants.][66183] These are still rejected when compiled, however they
```
Comment (Centril) [inline RELEASES.md]:
```suggestion
  enum variants.][66183] These are still rejected semantically, but
```

File: RELEASES.md [hunk 11]
```diff
@@ -1,3 +1,111 @@
...
+  can be seen and parsed by procedural macros.
```
Comment (Centril) [inline RELEASES.md]:
```suggestion
  can be seen and parsed by procedural macros and conditional compilation.
```

**Comment by pietroalbini** [comment]
We should also mention in the compatibility notes this is the last release with the current level of support for Tier 3 targets.

File: RELEASES.md (hunk 7 above)
Comment (XAMPPRocky) [reply RELEASES.md]: PRs linked mention testing. I haven't verified if it's one way or the other.

File: RELEASES.md [hunk 12]
```diff
@@ -1,3 +1,116 @@
...
+
+- [You can now specify `edition` in documentation code blocks to only test code
+  for that edition.][66238] E.g. `edition2018` will only test code when compiled
+  with the 2018 edition.
+
+Compatibility Notes
+-------------------
+
+- [Derive helper attributes for foreign derives no longer see or use other
+  derive's helper attributes.][64694]
```
Comment (petrochenkov) [inline RELEASES.md]: This description doesn't match what #64694 does.
Together with https://github.com/rust-lang/rust/pull/66529 the new behavior should be fully backward compatible with the old one, so this note should probably be just removed.

Comment (XAMPPRocky) [reply RELEASES.md]: I was pretty confused why it marked `relates` good to know.

File: RELEASES.md (hunk 7 above)
Comment (Mark-Simulacrum) [reply RELEASES.md]: Per https://doc.rust-lang.org/nightly/rustdoc/documentation-tests.html, "edition2018 tells rustdoc that the code sample should be compiled the 2018 edition of Rust. Similarly, you can specify edition2015 to compile the code with the 2015 edition." -- I believe I'm correct, this tests the code in 2018 edition regardless of `--edition` passed to rustdoc itself (i.e., your doc tests can have a different edition from the crate they're in).

**Review by ollie27 — COMMENTED** [review]
For rustdoc there were two more features stabilized that should be mentioned: #54733 and #61351

File: RELEASES.md [hunk 13]
```diff
@@ -1,3 +1,119 @@
...
+Misc
+----
+
+- [You can now specify `edition` in documentation code blocks to compile the block
+  for that edition.][66238] E.g. `edition2018` tells rustdoc that the code sample
+  should be compiled the 2018 edition of Rust.
+- [You can now provide custom themes to rustdoc with `--theme`, and check the
+  current theme with `--check-theme`.][54733]
+- [You can use `#[doc(cfg(item))]` to document that a item is only available with
+  certain features.][61351]
```
Comment (ollie27) [inline RELEASES.md]: It's actually only `#[cfg(doc)]` that was stabilized in the end. `#[doc(cfg(item))]` is still unstable.

`#[cfg(doc)]` is enabled in rustdoc when building documentation and finding doctests.

**Comment by pietroalbini** [comment]
We need to include this in the 1.41.0 stable branch, and it seems like there are no pending concerns in the PR.

@bors r+ rollup
r? @pietroalbini

**Pull request merged**